plugins/techdocs-node/src/stages/generate/types.ts

```diff
@@ -18,22 +18,62 @@ import { Entity } from '@backstage/catalog-model';
 import { Writable } from 'stream';
 import { Logger } from 'winston';
 import { ParsedLocationAnnotation } from '../../helpers';
-import { ContainerRunner } from '@backstage/backend-common';
+import {
+  ContainerRunner,
+  RunContainerOptions,
+} from '@backstage/backend-common';
 
-// Determines where the generator will be run
+/**
+ * Determines where the generator will be run. `'docker'` is a shorthand for running the generator in a container.
+ * If no {@link GeneratorOptions.containerRunner} is specified, the internal `DockerContainerRunner` will be used.
+ */
 export type GeneratorRunInType = 'docker' | 'local';
 
+/**
+ * Options passed to the {@link TechdocsContainerRunner.runContainer} method.
+ *
+ * Allows the deprecated `RunContainerOptions` for backward compatibility until it is removed.
+ *
+ * @public
+ */
+export type TechdocsRunContainerOptions =
+  | {
+      imageName: string;
+      command?: string | string[];
+      args: string[];
+      logStream?: Writable;
+      mountDirs?: Record<string, string>;
+      workingDir?: string;
+      envVars?: Record<string, string>;
+      pullImage?: boolean;
+      defaultUser?: boolean;
+    }
+  | RunContainerOptions;
+
+/**
+ * Handles the running of containers to generate TechDocs.
+ *
+ * Custom implementations, e.g. for Kubernetes or other execution environments, can be inspired by the internal default
+ * implementation `DockerContainerRunner`.
+ *
+ * Extends `ContainerRunner` for backward compatibility until it is removed.
+ *
+ * @public
+ */
+export interface TechdocsContainerRunner extends ContainerRunner {
```

Comment:
This types should not extends the `ContainerRunner` since it is deprecated and will be deleted. My recommendation is to copy the `ContainerRunner` props you need to the `TechdocsContainerRunner` you created.

Reply:
Hi @camilaibs,

it's summer time. I'm also just back from vacation.

Thanks for your review. I was trying to maintain backward compatibility until the `ContainerRunner` is finally removed. But you are right, that doesn't make sense here. Both interfaces share nearly the same declaration. Only `RunContainerOptions` has an optional `pullOptions` property that is not used any more and does not exist in the new `TechdocsContainerRunOptions` as in [the implementation of `DockerContainerRunner`](https://github.com/backstage/backstage/pull/25543/files#diff-b131b4ff78b97d217515ac30ebaaa3d2323e893a076189dbd5fced9c4515ed2cL41) before. I think there are more places where imports from backend-common can be removed or replaced by the Techdocs* types and interfaces introduced here.

- the [`TechdocsRunContainerOptions`](https://github.com/backstage/backstage/pull/25543/files#diff-e9ac315c875593eddfd73665103aeefb394528a2a6a2564870b808750cd323f8R331) may not need to allow `RunContainerOptions` as in the previous state of this PR
- the [static `fromConfig`](https://github.com/backstage/backstage/blob/master/plugins/techdocs-node/src/stages/generate/generators.ts#L45) method of the generator takes a `ContainerRunner` but may also take a `TechdocsContainerRunner`

I updated the PR accordingly. May I get another review?